### ISSUE TITLE
App doesn't work since Android 15

### ISSUE THREAD

**Issue by splattergamesextended** [jd1378/otphelper#141]
I don't know what happened, but the app doesn't do anything anymore. Sending a test notification works, but codes don't get copied or recognized anymore. There have also been no new codes in the history for about a week now.
I'm running GrapheneOS on the newest build (2024102400).
All permissions are set, I tried to deactivate the app, cleared cache, force quit and it changed nothing.

**Comment by jd1378** [comment]
duplicate of #140

the simple solution for now is to disable the new "Enhanced Notifications" from notification settings

the other solution that **I have not tested** but was suggested by [MishaalRahman](https://www.reddit.com/user/MishaalRahman/) if you know what `adb` is: 
```shell
adb shell cmd appops set --user 0 io.github.jd1378.otphelper RECEIVE_SENSITIVE_NOTIFICATIONS allow 
```
I'm pinning this since other people will be coming here to report the same issue probably.

**Comment by splattergamesextended** [comment]
Oh yeah, sorry that I missed that. Should have looked in the closed issues first. A permanent fix for other users would be amazing though! Thanks for the workaround, it finally works again!

**Comment by jd1378** [comment]
sadly this is not something I can fix, google decided to break this for all apps from android 15 going forward. this supposedly increases security in general, but I don't get it when the user's permission is already explicitly needed.

**Comment by splattergamesextended** [comment]
Aww man, I just found an article stating this:

"While digging through the [Android 14 QPR3 Beta 1](https://www.androidauthority.com/android-14-qpr3-beta-1-3411989/) update, I discovered the addition of a new permission named RECEIVE_SENSITIVE_NOTIFICATIONS. This permission has a protectionLevel of role|signature, which means it can only be granted to applications with the requisite role or to applications that the OEM signs. While the exact role that grants this permission hasn’t been defined yet, it’s likely that Google doesn’t intend to open this permission up to third-party apps."

So I can see how there will be no fix for it. Damn Google for this nonsense. If they just included reading OTP codes in AOSP, all of this wouldn't be an issue in the first place.

I guess the only "fix" is to notify users of this change within the app :( Thanks for your great work, it's of course Google that is to blame here, as always...

**Comment by jd1378** [comment]
yeah, I think telling the users about it inside the app is a good idea, I'll open this issue and close it after I get some time to implement it

thank you for the suggestion

**Comment by Dev-iL** [comment]
Is there any solution applicable to rooted devices? Can the app ask for root permissions, or use e.g. a magisk module to have the necessary access?

**Comment by jd1378** [comment]
maybe there is, but I am not aware of it
I don't own a rooted device, and for security reasons I rather keep it that way
so I probably won't actively look into this matter (I don't even have time to add small features right now), but I will try to review and accept PRs if anyone wants to add this

**Comment by leowood2000** [comment]
adb shell cmd appops set --user 0 io.github.jd1378.otphelper RECEIVE_SENSITIVE_NOTIFICATIONS allow 
I set it up but the app still not operated normal in my XIAOMI K80 PRO which is andriod 15  HyperOS 2.0
seems the SMS is not kind like the test SMS, it's a poped message show OTP number

**Comment by jd1378** [comment]
hmm
If a notification is not shown, then it won't work
I will have to implement a SMS-only mode some timer later when I have time
that's the only way to ensure this works without notifications

**Comment by leowood2000** [comment]
![Image](https://github.com/user-attachments/assets/0380867f-5d73-4ba6-b4ea-29745c95e82b)

The sms notification on xiaomi something like this, hyper os 2.0, can it be fixed? thank you

**Comment by jd1378** [comment]
I can't tell from the screenshot

if you get a normal android notification when a sms arrives, it should work
if you don't get a normal android notification, it won't work

just to be sure that android 15 is not messing with your notifications, have you tried disabling "Enhanced notifications" to see if it starts working or not?

and also, please send the text message that is not being detected on your phone, maybe it's a message that can't be detected by the app yet. replace the code inside with 123456 and remove sensitive information when posting here

**Comment by jd1378** [comment]
It seems it is working
Its probably the message
Please copy the message here, i cannot read or write chinese, i must be able to copy it

**Comment by leowood2000** [comment]
I finally got the solution, due to new Android features, apps with the "Read Notifications" permission are now required to separately request the "Read Sensitive Notifications" permission. just as you said,  the "enhanced notifications" should be disabled, However, this option cannot be found in Xiaomi settings.  so below following workaround can be used:

Download the "Create Shortcut" app.

Tap the three dots in the top right corner and check "Show system apps."

Search for "Settings" and tap on "Activity List."

In the activity selection, search for "Enhanced Notifications" and enter the "Details" page.

Disable "Enhanced Notifications."

So many thanks to you effort on this APP!!!

**Comment by jd1378** [comment]
Thanks for sharing your problem and the solution with everyone, it will help others that come here

**Comment by leowood2000** [comment]
my method using "create shortcut" app will be revoked  if I restart the phone, today I  double check your adb method, 

adb shell appops set --user 0 io.github.jd1378.otphelper RECEIVE_SENSITIVE_NOTIFICATIONS allow

it really worked.  thank you again!

Redmi K80pro  hyper OS 2.0

**Comment by jd1378** [comment]
That problem you are seeing is a different problem, and it's usually fixed by removing the app and installing again, it's an OS bug, not the program's fault

**Comment by Albirew** [comment]
ah indeed, after a reinstall, notifications are parsed again by the app

**Comment by ataurpc4005-source** [comment]
How do I add my number and receive the OTP?

**Comment by jd1378** [comment]
Please create a new issue
